packages/pwa/app/commerce-api/hooks/useBasket.js

```diff
@@ -54,18 +54,50 @@ export default function useBasket() {
              * to interact with a customer basket. All other calls are done through the
              * ShopperBaskets API, which in our case, uses OCAPI rather than commerce sdk.
              */
-            async getOrCreateBasket() {
+            async getOrCreateBasket(currency) {
                 const customerBaskets = await api.shopperCustomers.getCustomerBaskets({
                     parameters: {customerId: customer?.customerId}
                 })
 
                 if (Array.isArray(customerBaskets?.baskets)) {
-                    return setBasket(customerBaskets.baskets[0])
+                    let result
+
+                    result = setBasket(customerBaskets.baskets[0])
+
+                    // Update basket currency if necessary
+                    result =
+                        currency &&
+                        customerBaskets.baskets[0].currency !== currency &&
+                        this.updateBasketCurrency(currency, customerBaskets.baskets[0].basketId)
+
+                    return result
```

Comment:
There is a point of failure in this logic. In the scenario where you call `getOrCreateBasket()`, notice I didn't pass in a currency value, you'll make a request to get the customers baskets, set the state to be the first basket and then return result which is now assigned to `undefined`.

There are a couple things you can do here.

1. You can make currency a required param and throw and exception if it's undefined.
2. Change the signature of this function to include currency inside an options/opts param. Then clean up the logic as follows.


```
            async getOrCreateBasket(opts = {}) {
                const {currency} = opts
                const customerBaskets = await api.shopperCustomers.getCustomerBaskets({
                    parameters: {customerId: customer?.customerId}
                })
                
                // Throw if there was a problem getting the customer baskets
                isError(customerBaskets)) &&
                    throw new Error(customerBaskets)
                
                // We only support single baskets for now. Grab the first one. This will probably work as long as baskets is an array.. we might have to use `.?[0]`
                let basket = customerBaskets.baskets[0]
                
                if (!basket) {
                    // NOTE: Can we pass the currency in there?
                    basket = await api.shopperBaskets.createBasket({})
                    // NOTE: basket could be an error, you'll want to address that.
                    setBasket(basket)
                }
                
                 // Update basket currency if it was created with the wrong one, this will also set the state.
                if (basket.currency !== currency) {
                    this.updateBasketCurrency(currency, newBasket.basketId)
                }

                return basket
            },
```

Obviously I think the later is the better approach because we might support multiple baskets in the future, so you should see use passing more options like `id` to get a basket with a specific id, and the api doesn't require currency for getting or creating.

Reply:
Update, maybe we put the options on the hook, and not the individual function. This way `currency` will be available to all the function defined in the hook.